File: 206-reverse-linked-list/reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    if(head==NULL){
        return NULL;
    }
    struct ListNode *prev = head;
    struct ListNode *curr = head->next;
    while(curr!=NULL){
        struct ListNode *next = curr->next;
        curr->next = prev;
        prev = curr;
        curr = next;
    }

    head->next = NULL;
    return prev;
}

File: 206-reverse-linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null || head.next == null){
            return  head;
        }

        ListNode prev = head;
        ListNode curr = prev.next;
        while(curr != null){
            ListNode next = curr.next;

            curr.next = prev;

            prev = curr;
            curr = next;
        }

        head.next = null;
        return prev;
    }
}